### PR TITLE
feat(dli/job): add new spark job resource support

### LINKED ISSUE #1600: [DLI]Need a new resource to manage Spark jobs
For DLI service, we need a new resource to manage spark jobs.

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Support DLI Spark job submission resource.

**Which issue this PR fixes**:
fixes #1600

**Special notes for your reviewer**:

**Release note**:

```release-note
1. Add a new resource to support spark job submission:
2. Support the creation of dependent packages such as jar files and python files.
3. Add related document and test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDliSparkJobV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDliSparkJobV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDliSparkJobV2_basic
=== PAUSE TestAccDliSparkJobV2_basic
=== CONT  TestAccDliSparkJobV2_basic
--- PASS: TestAccDliSparkJobV2_basic (203.67s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       204.831s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDliSparkJobV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDliSparkJobV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDliSparkJobV2_basic
=== PAUSE TestAccDliSparkJobV2_basic
=== CONT  TestAccDliSparkJobV2_basic
--- PASS: TestAccDliSparkJobV2_basic (199.92s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       200.273s
```

File: docs/resources/dli_spark_job.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to submit a spark job.
+  If omitted, the provider-level region will be used.
+  Changing this parameter will submit a new spark job.
+
+* `queue_name` - (Required, String, ForceNew) Specifies the DLI queue name.
+  Changing this parameter will submit a new spark job.
+
+* `job_name` - (Required, String, ForceNew) Specifies the spark job name.
```
Comment (hmbig2) [inline docs/resources/dli_spark_job.md]: `name` is better

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+		// If the status of the spark job is "dead" or "success", it means that the life cycle of the job has ended.
+		if err == nil && resp != nil && (resp.State != batches.StateDead && resp.State != batches.StateSuccess) {
+			return fmt.Errorf("Spark job (%s) still exists.", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccApigApiPublishment_basic(name, dashName string) string {
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go]: pls rename the function name

File: docs/resources/dli_spark_job.md (hunk 1 above)
Comment (Lance52259) [reply docs/resources/dli_spark_job.md]: sure, this parameter has been updated.

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go]: done

File: docs/resources/dli_spark_job.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
+---
+subcategory: "Data Lake Insight (DLI)"
+---
+
+# huaweicloud_dli_spark_job
+
+Manages spark job resource of DLI  within HuaweiCloud
```
Comment (ShiChangkuo) [inline docs/resources/dli_spark_job.md]: two spaces between DLI and within

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDliSparkJobV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDliSparkJobV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDliSparkJobV2_basic
=== PAUSE TestAccDliSparkJobV2_basic
=== CONT  TestAccDliSparkJobV2_basic
--- PASS: TestAccDliSparkJobV2_basic (187.71s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       187.799s
```

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  cu_count   = 16
+  queue_type = "general"
+}
+
+%s
+
+resource "huaweicloud_dli_spark_job" "test" {
+  queue_name = huaweicloud_dli_queue.test.name
+  name       = "%s"
+  app_name   = "%s/simple_pyspark_test_DLF_refresh.py"
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go]: use the reference of the package resource

Comment (Lance52259) [reply huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_spark_job_test.go]: already updated.

File: docs/resources/dli_spark_job.md (hunk 3 above)
Comment (Lance52259) [reply docs/resources/dli_spark_job.md]: fixed

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccDliSparkJobV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccDliSparkJobV2_basic -timeout 360m -parallel 4
=== RUN   TestAccDliSparkJobV2_basic
=== PAUSE TestAccDliSparkJobV2_basic
=== CONT  TestAccDliSparkJobV2_basic
--- PASS: TestAccDliSparkJobV2_basic (182.07s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       182.205s
```

**Pull request merged**